module/x/gravity/client/rest/tx.go

```diff
@@ -72,7 +72,7 @@ func createValsetConfirmHandler(cliCtx client.Context, storeKey string) http.Han
 		}
 
 		cosmosAddr := cliCtx.GetFromAddress()
-		msg := types.NewMsgValsetConfirm(valset.Nonce, req.EthAddress, cosmosAddr, req.EthSig)
+		msg := types.NewMsgSubmitEthereumSignature(valset.Nonce, req.EthAddress, cosmosAddr, req.EthSig)
```

Comment:
This is where we are swapping specialized methods for generic methods dealing with interfaces and 'any's right? If so, I would take another look at the use of the nonce here. It is a part of the valset content and should probably not be exposed to the generic method. Other EthereumTx's do not have nonces which are semantically the same. TransferBatch and External contract call nonces do not have a single global sequence. Including this value in a generic struct or method is probably wrong.